sklearn/linear_model/stochastic_gradient.py

```diff
@@ -355,15 +345,15 @@ def _partial_fit(self, X, y, alpha, C,
                                                            self.classes_, y)
         sample_weight = self._validate_sample_weight(sample_weight, n_samples)
 
-        if self.coef_ is None or coef_init is not None:
+        if getattr(self, "coef_", None) is None or coef_init is not None:
```

Comment:
I tried to replace this with `hasattr` but there are multiple places in the scikit-learn code that sets `coef_` to None. I can try again to see how hard it is to replace `getattr` with `hasattr`.